.github/workflows/msvc-ci.yml

```diff
@@ -150,4 +150,271 @@ jobs:
+              Join-String -Separator ([System.Environment]::NewLine);
+
+            $pkgconfig_file = Join-Path -Path $luajit_pkgconfig_dir -ChildPath "${pkgconfig_filename}.pc";
+            Set-Content $pkgconfig_file $pkgconfig_content -NoNewLine;
+          }
+
+          $luajit_target_dir_unix = & C:\msys64\usr\bin\cygpath -u "$luajit_target_dir";
+          $luajit_tar_gz = Join-Path -Path "${{ runner.temp }}" -ChildPath "luajit.tar.gz";
+          $luajit_tar_gz_unix = & C:\msys64\usr\bin\cygpath -u "$luajit_tar_gz";
+
+          & C:\msys64\usr\bin\bash -lc "tar -C ""$luajit_target_dir_unix"" -czvf ""$luajit_tar_gz_unix"" .";
+
+          Remove-Item -Path $luajit_target_dir -Recurse;
+
+          Add-Content "${{ github.env }}" "LUAJIT_TAR_GZ=${luajit_tar_gz}";
+          Add-Content "${{ github.env }}" "LUAJIT_TAR_GZ_UNIX=${luajit_tar_gz_unix}";
+      
+      - name: Extract LuaJIT tar gz on GTK's directory
+        run: |
+          $gtk_dir_unix = & C:\msys64\usr\bin\cygpath -u "${{ env.GTK_DIR }}";
+
+          & C:\msys64\usr\bin\bash -lc "tar -C ""$gtk_dir_unix"" -xzvf ""${{ env.LUAJIT_TAR_GZ_UNIX }}""";
+
+      - name: Setup Python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+      
+      - name: Install meson
+        run: pip install meson
+
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Configure lgi through meson
+        run: meson setup lgi-build . --prefix "${{ env.GTK_DIR }}" "-Dlua-bin=luajit" -Dtests=false
+      
+      - name: Build lgi
+        run: meson compile -C lgi-build
+      
+      - name: Install lgi
+        run: meson install -C lgi-build
+
+      - name: Create a simple test script using lgi
+        run: |
+          $test_script = @'
+          local lgi = assert(require("lgi"))
+          local Gtk = assert(lgi.require("Gtk", "${{ matrix.gtk-major-version }}.0"))
+
+          local app = Gtk.Application({ application_id = "org.lgi-devs.lgi" })
+
+          function app:on_activate()
+              local w = Gtk.ApplicationWindow()
+              w:set_default_size(900, 600)
+              w:set_title("My great title")
+
+              w.application = self
+
+              if ("${{ matrix.gtk-major-version }}" == "3") then
+                  w:show_all()
+              elseif ("${{ matrix.gtk-major-version }}" == "4") then
+                  w:present()
+              else
+                  error("Unknown GTK version")
+              end
+
+              w:close()
+          end
+
+          app:run()
+          '@;
+          
+          $test_file = Join-Path -Path "${{ runner.temp }}" -ChildPath "small-lgi-test.lua";
+
+          # Write the script above to file
+          Set-Content $test_file $test_script -NoNewLine;
+
+          # Set the test file name on LGI_TEST_FILE environment variable
+          Add-Content "${{ github.env }}" "LGI_TEST_FILE=${test_file}";
```

Comment:
> Now that this script is present twice in `msvc-ci.yml`... could it be moved to an extra test file and only referenced?
> 
> (Alternaticely, instead of inventing a new test suite, just the usual LGI test suite could be run... but I assume you have a reason against it...)

oh, this script is left there in text, because all "${{ }}" gets replaced by GitHub's own mechanisms. However, I could place such scripts as files (maybe one for gtk 3, and another for gtk 4), and call them accordingly.

Honestly, I didn't look at tests folder, because I don't have a deeper understanding of lgi. I (possibly mistaken) assumed that -Dtests=true did something special, but I'll look each script on tests folder and see if I can run them locally without much of a hassle.

Reply:
>  However, I could place such scripts as files (maybe one for gtk 3, and another for gtk 4), and call them accordingly.

Random idea: Pass the gtk 3 / 4 info as a command line argument and use that in the script for the `if`.

Random idea 2: Auto-detect the GTK version. Something like `if w.show_all then w:show_all() elseif w.present then w:present() else error()`. Perhaps with a comment in there explaining which GTK version is which.

Thanks for giving this a try!

>  (possibly mistaken) assumed that -Dtests=true did something special

All it does is to tell meson to skip the `tests/` folder. So, basically it skips all of `tests/meson.build`.

That directory generates a special GIR for tests that is provided by gobject-introspection. The result is made available to `tests/test.lua` via environment variables and run... but not on Windows (dunno why), so that seems unrelated to this PR. But this is actually the main test suite. :-(  `test.lua` executes all the `*.lua` files in `tests/`.

The other thing is compiling and running `tests/test_c.c`, which apparently I added at five years for a special case. I do not remember anything. :see_no_evil: 

Anyway.. I guess this means you can keep `-Dtests=false` since there is already a "skip the test suite on Windows" in there.